backend/src/src-licenseinfo/src/test/java/com/siemens/sw360/licenseinfo/parsers/SPDXParserTest.java

```diff
@@ -0,0 +1,138 @@
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.spdx.rdfparser.model.SpdxDocument;
+import org.spdx.rdfparser.SPDXDocumentFactory;
+
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.Optional;
+
+import static com.siemens.sw360.licenseinfo.TestHelper.*;
+import static com.siemens.sw360.licenseinfo.parsers.SPDXParser.FILETYPE_SPDX_EXTERNAL;
+import static com.siemens.sw360.licenseinfo.parsers.SPDXParser.FILETYPE_SPDX_INTERNAL;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.*;
+
+/**
+ * @author: maximilian.huber@tngtech.com
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class SPDXParserTest {
+
+    private SPDXParser parser;
+
+    private AttachmentContentStore attachmentContentStore;
+
+    @Mock
+    private AttachmentConnector connector;
+
+    public static final String spdxExampleFile = "SPDXRdfExample-v2.0.rdf";
+    private static String baseUri;
+
+    @Before
+    public void setUp() throws Exception {
+        attachmentContentStore = new AttachmentContentStore(connector);
+
+        parser = new SPDXParser(connector, attachmentContentStore.getAttachmentContentProvider());
+
+        attachmentContentStore.put(spdxExampleFile);
+    }
+
+    private void assertIsResultOfExample(LicenseInfo result){
+        assertLicenseInfo(FILETYPE_SPDX_EXTERNAL, result);
+
+        assertThat(result.getFilenames().size(), is(1));
+        assertThat(result.getFilenames().get(0), is(spdxExampleFile));
+
+        assertThat(result.getLicenseTextsSize(), is(5));
+        assertThat(result.getLicenseTexts().stream()
+                .map(l -> l.contains("\"THE BEER-WARE LICENSE\""))
+                .filter(b -> b)
+                .findAny()
+                .get(),
+                is(true));
+        assertThat(result.getCopyrightsSize(), is(1));
+        assertThat(result.getCopyrights(), hasItem("Copyright 2008-2010 John Smith"));
+    }
+
+    @Test
+    public void testIsApplicableTo() throws Exception {
+        try {
+            SPDXParser.ACCEPTABLE_ATTACHMENT_TYPES.stream()
+                    .forEach(attachmentType -> SPDXParser.ACCEPTABLE_ATTACHMENT_FILE_EXTENSIONS.stream()
+                            .forEach(extension -> {
+                                String filename = "filename." + extension;
+                                try {
+                                    attachmentContentStore.put(filename, "");
+                                } catch (SW360Exception e) {
+                                    throw new UncheckedSW360Exception(e);
+                                }
+                                Attachment attachment = makeAttachment(filename, attachmentType);
+                                try {
+                                    assertThat(parser.isApplicableTo(attachment), is(true));
+                                } catch (TException e) {
+                                    e.printStackTrace();
```

Comment:
Yes, you are right, this looks like an autogenerated catch, which was not corrected.